Bottom: 94f7360bf39acd1dc322e495f7d49bc02f64e57e
Top:    d5ef12c8001b374eb85c8f7c27e79892812b6c7f
Author: Alexandre Oliva <aoliva@redhat.com>
Date:   2016-06-04 02:56:56 -0300

Refresh of remaining-unary-exprs.patch

---

diff --git a/libcc1/libcp1plugin.cc b/libcc1/libcp1plugin.cc
index af22bb3..89ddf15 100644
--- a/libcc1/libcp1plugin.cc
+++ b/libcc1/libcp1plugin.cc
@@ -2633,7 +2633,7 @@ plugin_unary_type_expr (cc1_plugin::connection *self,
 
       // We're using this for sizeof...(pack).  */
     case TYPE_PACK_EXPANSION:
-      result = make_pack_expansion (op0);
+      result = make_pack_expansion (type);
       PACK_EXPANSION_SIZEOF_P (result) = true;
       break;
 
@@ -2644,7 +2644,7 @@ plugin_unary_type_expr (cc1_plugin::connection *self,
   if (template_dependent_p)
     processing_template_decl--;
 
-  return convert_out (ctx->preserve (val));
+  return convert_out (ctx->preserve (result));
 }
 
 gcc_expr
